.github/workflows/_charm-release.yaml

```diff
@@ -136,27 +142,22 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 1
-      - name: Select charmhub channel
-        uses: canonical/charming-actions/channel@2.1.1
-        id: channel
       - name: Fetch charm artifacts
         uses: actions/download-artifact@v3
         with:
           name: charms
           path: "${{ github.workspace }}/${{ inputs.charm-path }}"
-      - name: Upload charm to charmhub
-        uses: canonical/charming-actions/upload-charm@2.5.0-rc
-        with:
-          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
-          github-token: "${{ secrets.GITHUB_TOKEN }}"
-          channel: "${{ steps.channel.outputs.name }}"
-          built-charm-path: "${{ matrix.path }}"
-          charm-path: "${{ inputs.charm-path }}"
-          tag-prefix: "${{ inputs.release-tag-prefix }}"
-          charmcraft-channel: "${{ inputs.charmcraft-channel }}"
-          # We set destructive mode to false, otherwise runner's OS would have to match
-          # charm's 'build-on' OS.
-          destructive-mode: false
+      - name: Upload charm to Charmhub
```

Comment:
I have, but I disagree on the benefits.

I think being able to see the logic is a big advantage: no one in the team knows exactly what charming-actions do for this exact reason. 

We already have re-usable workflows that abstract the work: this very file is not even used directly, so if you want a high-level view there's `charm-release.yaml` to look at; if you're opening this, you arguably want to know what's happening.

I'd also argue that looking at a few lines of bash is a lot simpler than having another level of nesting (`release.yaml` in charms calls `charm-release.yaml`, which calls `_charm_release.yaml`, which should call yet another `upload` action).

Reply:
About testing, it's true you can't unit-test these actions as a whole. I'd argue that for that, I'd rather have a fake repository where we run these workflows live, and we test new changes there instead.

The `noctua` code **is** unit-tested though. Once we have the automation in place, we can get automated version bumping of that tool instead. Renovate is useful only if we have a few requirements:
- we have a way to test the new action **in the PR**, which doesn't make much sense to me; we don't want to test when updating the CI in Prometheus/Grafana/all of our charms, we want to test before merging in `canonical/observability`;
- the automatic change saves us PR work; here, having this job as an action only produces one PR (to this repo, assuming the actions live somewhere else; otherwise you'd need to remember to bump manually); the CI on the actual charms would pick up a new tag on the whole `release` workflow, which would happen regardless

Reply:
Re using composite actions, agreed it is subjective.  I like the benefits of:
* calling an action with a name that state's its purpose
* being able to test the entire step, not just the noctua portion

and I don't mind paying for that with an extra layer of abstraction.  But there's no right or wrong

Reply:
as for testing, we should have *something* more than our production repos being the first recorded result.  There's options to make that work (this repo could trigger a PR on another repo, or at least we run this branch through a few repos and link them in the PR description)